### PR TITLE
fix stateful reward carry

### PR DESCRIPTION
**Pull request by budzianowski**
Do we expect the `initial_carry` to output for the whole trajectory carries?

### CODE REVIEW

File: ksim/task/rl.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def get_rewards(
         reward_carry = rewards_carry[reward_name]
 
         if isinstance(reward, StatefulReward):
-            reward_carry = jnp.where(trajectory.done, reward.initial_carry(rng), reward_carry)
+            reward_carry = jnp.where(trajectory.done[-1], reward.initial_carry(rng), reward_carry)
```
Comment (codekansas) [inline ksim/task/rl.py]: nit
```suggestion
            reward_carry = jnp.where(trajectory.done[..., -1], reward.initial_carry(rng), reward_carry)
```

**Review by codekansas — APPROVED** [review]
